src/assets/CollectiblesController.test.ts

```diff
@@ -1375,168 +1375,138 @@ describe('CollectiblesController', () => {
+    const mockTransactionId = '60d36710-b150-11ec-8a49-c377fbd05e27';
 
-      const collectible = {
-        address: '0x02',
-        tokenId: '1',
-        name: 'name',
-        image: 'image',
-        description: 'description',
-        standard: 'standard',
-        favorite: false,
-      };
+    const mockCollectible = {
+      address: '0x02',
+      tokenId: '1',
+      name: 'name',
+      image: 'image',
+      description: 'description',
+      standard: 'standard',
+      favorite: false,
+    };
 
-      collectiblesController.updateCollectibleByAddressAndTokenId(
-        collectible.address,
-        collectible.tokenId,
-        { ...collectible, transactionId: mockTransactionId },
-      );
+    const expectedMockCollectible = {
+      address: '0x02',
+      description: 'description',
+      favorite: false,
+      image: 'image',
+      name: 'name',
+      standard: 'standard',
+      tokenId: '1',
+      transactionId: mockTransactionId,
+    };
 
+    it('should return undefined when the collectible does not exist', async () => {
       expect(
         collectiblesController.findCollectibleByAddressAndTokenId(
-          collectible.address,
-          collectible.tokenId,
-        ),
-      ).toBe(undefined);
+          mockCollectible.address,
+          mockCollectible.tokenId,
+        )?.collectible,
+      ).toBeUndefined();
     });
 
     it('should return the collectible by the address and tokenId with the prop transactionId', async () => {
-      const mockTransactionId = '60d36710-b150-11ec-8a49-c377fbd05e27';
-
-      const collectible = {
-        address: '0x02',
-        tokenId: '1',
-        name: 'name',
-        image: 'image',
-        description: 'description',
-        standard: 'standard',
-        favorite: false,
-      };
-
       await collectiblesController.addCollectible(
-        collectible.address,
-        collectible.tokenId,
-        collectible,
+        mockCollectible.address,
+        mockCollectible.tokenId,
+        mockCollectible,
       );
 
-      collectiblesController.updateCollectibleByAddressAndTokenId(
-        collectible.address,
-        collectible.tokenId,
-        { ...collectible, transactionId: mockTransactionId },
-      );
+      collectiblesController.updateCollectibleByAddressAndTokenId({
+        ...mockCollectible,
+        transactionId: mockTransactionId,
+      });
 
       expect(
         collectiblesController.findCollectibleByAddressAndTokenId(
-          collectible.address,
-          collectible.tokenId,
-        ),
-      ).toStrictEqual({
-        address: '0x02',
-        description: 'description',
-        favorite: false,
-        image: 'image',
-        name: 'name',
-        standard: 'standard',
-        tokenId: '1',
-        transactionId: '60d36710-b150-11ec-8a49-c377fbd05e27',
-      });
+          mockCollectible.address,
+          mockCollectible.tokenId,
+        )?.collectible,
```

Comment:
Same for this, and other examples.

Reply:
Here I need the`?.collectible` because the object is possibly null and it returns `collectible: Collectible, index: number`

Reply:
Ah... hmm. I think I see why: you're using `addCollectible` and `updateCollectibleByAddressAndTokenId` to put a collectible in the list of collectibles, and what you're thinking is that that may not work as you expect, so you need the `?.` here. That said, do you need to call these methods in order to test `findCollectibleByAddressAndTokenId`? `findCollectibleByAddressAndTokenId` doesn't really care how the collectibles ends up in the state, after all. What if you do something simpler that you know will work, like a) update the state of the controller directly so that `allCollectibles` takes the shape you want to satisfy the test, or b) when you make a new instance of `controller`, pass in a default state object that takes the shape you want to satisfy the test?

Reply:
Thank you so much for the review!
You were completely right, I choose to update the state directly and now they are more isolated and precise.
It's way better now